### DISCUSSION TITLE
Lesson 12: Reentrancy Guard/OpenZeppelin Problems

### DISCUSSION THREAD

**Discussion by DukeLurkin** [Cyfrin/foundry-full-course-cu#1209] [Q&A]
Looks like something is wrong with the compatibility between solidity and openzeppelin.  When I tried to run forge build the first time I got this:
<img width="1176" alt="ReentrancyGuardBad" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/7d72ed61-4ad7-480e-ace1-fc3484cdab37">

I've tried to install older versions of zep and that didn't work.  Then I tried to change the folder path to how Zeppelin is now structured(there is no security folder, ReentrancyGuard is in Utils now) and I got a this error:
<img width="717" alt="solidityVersionBad" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/612921e8-dafd-43d0-b653-9630ac3222c9">

Then I updated pragma to ^0.8.20 from .19 and now !success is not recognized. 
<img width="704" alt="badSuccess" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/706a0f4a-d7cf-46e1-b897-2a6da53339d2">

 Seems like the more I try to update the worse it gets.  Any suggestions?

**Comment by ghost** [comment]
`success` it's not defined, you need something like this:

```javascript
bool success = IERC20(tokenCollateralAddress).transferFrom(msg.sender, address(this), amountCollateral);
```

**Comment by YourDailyBlockchain** [comment]
Did you ever get this to work? I'm having the same issue. Thanks

**Reply by YourDailyBlockchain** [reply]
I got it to work.
I went into the Lib folder and deleted my Zeppelin contracts, then installed this version of OZ:
forge install openzeppelin/openzeppelin-contracts@v4.8.3 --no-commit
Ran forge build and it compiled.

**Reply by JR-Vickers** [reply]
I had the same issue.  What worked for me was double-checking my foundry.toml file - there was a small typo in my remappings.  Make sure it looks like this:

```
remappings = ["@chainlink/contracts=lib/chainlink-brownie-contracts/contracts/", '@openzeppelin/contracts=lib/openzeppelin-contracts/contracts']
```

**Comment by tobezhanabi** [comment]
This should work
